py/autoevals/moderation.py

```diff
@@ -0,0 +1,59 @@
+from braintrust_core.score import Score, Scorer
+
+from .oai import run_cached_request
+
+REQUEST_TYPE = "moderation"
+
+
+class Moderation(Scorer):
+    """
+    A scorer that uses OpenAI's moderation API to determine if AI response contains ANY flagged content.
+    """
+
+    threshold = None
+    extra_args = {}
+
+    def __init__(self, threshold=None, api_key=None, base_url=None):
+        """
+        Create a new Moderation scorer.
+
+        :param threshold: Optional. Threshold to use to determine whether content has exceeded threshold. By
+        default, it uses OpenAI's default. (Using `flagged` from the response payload.)
+        :param api_key: OpenAI key
+        :param base_url: Base URL to be used to reach OpenAI moderation endpoint.
+        """
+        self.threshold = threshold
+
+        self.extra_args = {}
+
+        if api_key:
+            self.extra_args["api_key"] = api_key
+        if base_url:
+            self.extra_args["base_url"] = base_url
+
+    def _run_eval_sync(self, output, __expected=None):
```

Comment:
mind adding the async variant too?

Reply:
It can be a bit annoying but here's a pattern we use: https://github.com/braintrustdata/autoevals/pull/60/files#diff-b0876149464a4294b532c23f50f8db0143392b1f8eb2ce9b8e9d1c5cdca0e06bR214

Reply:
* [For my context] By default, `_run_eval_async` calls `_run_eval_sync` per https://github.com/braintrustdata/braintrust-sdk/blob/main/core/py/src/braintrust_core/score.py#L52 - Is this more like a standard pattern in the code base rather than functionality?
* [Out of scope for this PR] Wonder if it makes sense to mark `_run_eval_async` as abstract as well, so all children will need to override it.

Reply:
Added an async variant: https://github.com/braintrustdata/autoevals/pull/63/commits/10a36d9785a7428d7c8db3a9385367d10282dae6#diff-059564dc34fe7366598e59c4b9fbcab770ba4689f9c9418010178055baed560eR44